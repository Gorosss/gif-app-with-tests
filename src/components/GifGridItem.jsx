import React, {useState} from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {

    const [copied, setCopied] = useState(false);


    const copyToClipboard = (text) => {
        // Creamos un elemento textarea para copiar el texto
        const textarea = document.createElement('textarea');
        // Asignamos el texto que queremos copiar al valor del textarea
        textarea.value = text;
        // Hacemos que el textarea sea invisible
        textarea.style.position = 'fixed';
        textarea.style.top = '0';
        textarea.style.left = '0';
        textarea.style.opacity = '0';
        // AÃ±adimos el textarea al DOM
        document.body.appendChild(textarea);
        // Seleccionamos el texto dentro del textarea
        textarea.select();
        // Ejecutamos el comando de copiar
        document.execCommand('copy');
        // Removemos el textarea del DOM
        document.body.removeChild(textarea);

        setCopied(true);
        setTimeout(() => {
          setCopied(false);
        }, 2000);
      };

    const handleClickCard = (url) => {
        copyToClipboard(url)
    }

    return (
        <div className="card animate__animated animate__fadeIn" onClick={() => handleClickCard(url)}>
            {copied && <div style={{ position: 'fixed', transform: 'translateX(-50%)', background: 'rgba(0, 0, 0, 0.8)', color: '#fff', padding: '10px', borderRadius: '5px' }}>Copied!</div>}

            <img src={ url } alt={ title } />
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
