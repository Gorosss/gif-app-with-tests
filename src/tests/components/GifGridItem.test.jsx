import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('Test <GifGridItem />', () => {

    const title = 'Title';
    const url   = 'https://localhost';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    
    test('It must show the component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('It must have a title in a p', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );

    })

    test('It must have the image matching the URL and alt props', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );

    });


    test('It must have animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe( true );

    })
    
    
    
    

})
