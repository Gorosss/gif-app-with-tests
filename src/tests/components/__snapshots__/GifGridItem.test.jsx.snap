// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe de mostrar el componente correctamente 1`] = `ShallowWrapper {}`;

exports[`Test <GifGridItem /> It must show the component correctly 1`] = `ShallowWrapper {}`;
